cli/index.ts

```diff
@@ -359,11 +364,13 @@ const builtYargs = createYargsCli({
           process.on("SIGINT", () => {
             watcher.close();
             watching = false;
+            process.exit(1);
           });
           while (watching) {
             await new Promise((resolve, reject) => setTimeout(() => resolve(), 100));
           }
         }
+        process.exit(0);
```

Comment:
i'm not sure this belongs here. this code should never really execute if `watching === true`. i would expect instead a `return 0` if `watching === false`, before we enter the `if (watching)` block.

in fact, that `return` should really return a 0 or 1 as according to whether or not the `CompiledGraph` contains errors: https://source.cloud.google.com/tada-analytics/github_dataform-co_dataform/+/master:protos/core.proto;l=235?q=compiledgraph

Reply:
Now records the result of `compiledGraphHasErrors` and returns according to the result of that if watching is not set